emcc.py

```diff
@@ -2320,6 +2319,24 @@ def do_binaryen(final, target, asm_target, options, memfile, wasm_binary_target,
           passes.append('minifyWhitespace')
         final = shared.Building.js_optimizer_no_asmjs(final, passes)
       if DEBUG: save_intermediate('postclean', 'js')
+  # replace placeholder strings with correct subresource locations
+  if shared.Settings.SINGLE_FILE:
+    f = open(final, 'r')
+    js = f.read()
+    f.close()
+    f = open(final, 'w')
+    for tup in [
```

Comment:
how about `for target, to_replace, condition in [` (maybe with better variable names, but you see what I mean, to avoid `tup[1]` etc.)